src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -200,6 +201,32 @@ abstract class AbstractServerTask extends AbstractTask {
         }
     }
 
+    protected void copyConfigDirectory() {
+        //merge default server.env with one in config directory
+        if(server.configDirectory != null && server.appendServerEnv) {
```

Comment:
Not sure why you are checking if `server.configDirectory != null` here. The code checked that `server.configDirectory.exists()` before calling this method. Also, I think you should do line 210 before this if statement and change this if statement to the following:

`if (configDirServerEnv.exists() && server.appendServerEnv)`